### PR TITLE
chore: add cypress

### PR DESCRIPTION
**Pull request by Lukas742**
Todo:

- [x] coverage
- ~~matrix testing for different React versions~~ (later)

### CODE REVIEW

File: cypress/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "target": "ES2015",
+    "lib": ["es2019", "dom"],
+    "types": ["cypress", "node"]
```
Comment (MarcusNotheis) [inline cypress/tsconfig.json:5]: Please include`@testing-library/cypress` as well:
```suggestion
    "types": ["cypress", "node", "@testing-library/cypress"]
```

File: package.json [hunk 2]
```diff
@@ -12,6 +12,8 @@
     "postinstall": "rimraf node_modules/@types/mocha",
     "pretest": "rimraf coverage && lerna run build:i18n",
     "test": "jest --config=config/jest.config.js --coverage",
+    "test:cypress": "lerna run build && cypress run --component && lerna run clean",
+    "test:all": "yarn test:cypress && yarn test",
     "clean": "lerna run clean --stream && rimraf coverage",
```
Comment (MarcusNotheis) [inline package.json:17]: I would add another `"cypress:open": "cypress open"` script here

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -79,7 +92,6 @@ jobs:
       - name: run eslint
         run: yarn lint
 
-
   deploy-next-docs:
     needs: ['test-and-build']
```
Comment (MarcusNotheis) [inline .github/workflows/main.yml:96]:
```suggestion
    needs: ['test-jest-and-build']
```

File: vite.config.ts.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// vite.config.ts
+import { defineConfig } from 'vite';
```
Comment (MarcusNotheis) [inline vite.config.ts.js:2]: please delete this file

File: vite.config.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { defineConfig, searchForWorkspaceRoot } from 'vite';
+import { dirname, join } from 'path';
```
Comment (MarcusNotheis) [inline vite.config.ts:4]: some unused imports?

**Review by MarcusNotheis — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Lukas742) [reply package.json:18]: Done. I set component testing and Chrome as default.

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**